tests/cc/test_const_prop.cq

```diff
@@ -0,0 +1,102 @@
+# File:         test_const_prop.cq
+# Purpose:      test cQasm constant propagation
+# Notes:        most constructs make no practical sense
+
+version 1.2
+qubits 17
+
+pragma @ql.name("test_const_prop")
+pragma @ql.platform("config_cc_s17_direct_iq_openql_0_10.json")
+
+.def
+map qInput1 = q[0]
+map qInput2 = q[1]
+map qAncilla3 = q[2]
+map qOutput4 = q[3]
+
+# var mAncilla3: bool   encountered incompatible object reference to mAncilla3 (size=0)
+map mAncilla3 = b[2]
+
+map i = creg(0)
+map j = creg(1)
+
+.condConstExpr
+while (4==4) {
+}
+
+.deadCodeElim
+if (3<2) {
```

Comment:
Why aren't these constant expressions already folded by libqasm?

Reply:
Quoting from `arch::cc::pass::gen::vq1asm::detail::codegen.h`, especially see that last sentence:

   /**
     * Handle expressions.
     *
     * To understand how cQASM functions end up in the IR, please note that functions are handled during analyzing
     * cQASM, see 'AnalyzerHelper::analyze_function()'.
     *
     * A default set of functions that only handle constant arguments is provided by libqasm, see
     * 'register_into(resolver::FunctionTable &table)'. These functions add a constant node to the IR when called (and
     * fail if the arguments are not constant)
     *
     * Some of these are overridden by OpenQL to allow use of non-constant arguments. This is a 2 step process, where
     * 'convert_old_to_new(const compat::PlatformRef &old)' adds functions to ir->platform using 'add_function_type',
     * and 'ql::ir::cqasm:read()' then walks 'ir->platform->functions' and adds the functions using
     * 'register_function()'. These functions add a 'cqv::Function' node to the IR (even if the arguments are constant,
     * so overriding a function defeats libqasm's constant removal for that function).
     */

Reply:
Created an issue https://github.com/QuTech-Delft/OpenQL/issues/451 for this